qp_shotgun/qc_filter/qc_filter.py

```diff
@@ -0,0 +1,314 @@
+            return False, error_msg
+
+    return True, ""
+
+
+def _per_sample_ainfo(out_dir, samples, fwd_and_rev=False):
+    files = []
+    missing_files = []
+
+    suffixes = ['%s.R1.trimmed.filtered.fastq.gz',
+                '%s.R2.trimmed.filtered.fastq.gz']
+
+    for _, rp, _, _ in samples:
+        smd = partial(join, out_dir)
+        for suff in suffixes:
+            fname = smd(suff % rp)
+            if exists(fname):
+                files.append(fname)
+            else:
+                missing_files.append(fname)
+
+    if not files:
+        # QC_Filter did not create any files, which means that no sequence
+        # was kept after filteration
+        raise ValueError("No sequences left after filtering")
+
+    return [ArtifactInfo('QC_Trim files', 'per_sample_FASTQ', files)]
+
+
+def qc_filter(qclient, job_id, parameters, out_dir):
+    """Run filtering using Bowtie2 with the given parameters
+
+    Parameters
+    ----------
+    qclient : tgp.qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values to run split libraries
+    out_dir : str
+        The path to the job's output directory
+
+    Returns
+    -------
+    bool, list, str
+        The results of the job
+    """
+    # Step 1 get the rest of the information need to run Bowtie2
+    qclient.update_job_step(job_id, "Step 1 of 4: Collecting information")
+    artifact_id = parameters['input']
+    del parameters['input']
+
+    # Get the artifact filepath information
+    artifact_info = qclient.get("/qiita_db/artifacts/%s/" % artifact_id)
+    fps = artifact_info['files']
+
+    # Get the artifact metadata
+    prep_info = qclient.get('/qiita_db/prep_template/%s/'
+                            % artifact_info['prep_information'][0])
+    qiime_map = prep_info['qiime-map']
+
+    # Step 2 generating command
+    qclient.update_job_step(job_id, "Step 2 of 4: Generating"
+                                    " QC_Filter commands")
+    # Creating temp directory for intermediate files
+    temp_dir = mkdtemp()
+
+    rs = fps['raw_reverse_seqs'] if 'raw_reverse_seqs' in fps else []
+    commands, samples = generate_qc_filter_commands(fps['raw_forward_seqs'],
+                                                    rs, qiime_map, out_dir,
+                                                    temp_dir, parameters)
+
+    # Step 3 execute filtering command
+    len_cmd = len(commands)
+    msg = "Step 3 of 4: Executing QC_Trim job (%d/{0})".format(len_cmd)
+    success, msg = _run_commands(qclient, job_id, commands, msg)
+    if not success:
+        return False, None, msg
+
+    # Step 4 generating artifacts
+    msg = "Step 4 of 4: Generating new artifacts (%d/{0})".format(len_cmd)
+    success, msg = _run_commands(qclient, job_id, commands, msg)
+    ainfo = _per_sample_ainfo(out_dir, samples, bool(rs))
+
+    rmtree(temp_dir)

```

Comment:
A preferred method might be to invoke maketemporarydirectory under a `with` statement, per [this](https://stackoverflow.com/a/22726782) StackOverflow comment. This will automatically clean up the temp dir if an exception breaks execution prior to getting to the `rmtree` statement.